src/interchange/src/avro.rs

```diff
@@ -945,27 +1000,67 @@ impl Encoder {
         }
 
         let writer_schema = build_schema(&columns);
+        let consistency_schema = get_consistency_schema();
         Encoder {
             columns,
             writer_schema,
+            consistency_schema,
         }
     }
 
     pub fn writer_schema(&self) -> &Schema {
         &self.writer_schema
     }
 
-    pub fn encode_unchecked(&self, schema_id: i32, diff_pair: DiffPair<&Row>) -> Vec<u8> {
+    pub fn consistency_schema(&self) -> &Schema {
+        &self.consistency_schema
+    }
+
+    pub fn encode_unchecked(
+        &self,
+        schema_id: i32,
+        diff_pair: DiffPair<&Row>,
+        transaction_id: Option<String>,
+    ) -> Vec<u8> {
         let mut buf = Vec::new();
         buf.write_u8(0).expect("writing to vec cannot fail");
         buf.write_i32::<NetworkEndian>(schema_id)
             .expect("writing to vec cannot fail");
-        let avro = self.diff_pair_to_avro(diff_pair);
+        let avro = self.diff_pair_to_avro(diff_pair, transaction_id);
         debug_assert!(avro.validate(self.writer_schema.top_node()));
         avro::encode_unchecked(&avro, &self.writer_schema, &mut buf);
         buf
     }
 
+    pub fn encode_transaction_unchecked(
+        &self,
+        schema_id: i32,
+        id: &str,
+        status: &str,
+        message_count: Option<i64>,
+    ) -> Vec<u8> {
+        let mut buf = Vec::new();
+        buf.write_u8(0).expect("writing to vec cannot fail");
+        buf.write_i32::<NetworkEndian>(schema_id)
+            .expect("writing to vec cannot fail");
+
+        let transaction_id = Value::String(id.to_owned());
+        let status = Value::String(status.to_owned());
+        let event_count = match message_count {
+            None => Value::Union(0, Box::new(Value::Null)),
+            Some(count) => Value::Union(1, Box::new(Value::Long(count))),
+        };
+
+        let avro = Value::Record(vec![
+            ("id".into(), transaction_id),
+            ("status".into(), status),
+            ("event_count".into(), event_count),
+        ]);
+        debug_assert!(avro.validate(self.consistency_schema.top_node()));
+        avro::encode_unchecked(&avro, &self.consistency_schema, &mut buf);
+        buf
+    }
```

Comment:
I don't think any of this belongs on the `Encoder`? The consistency schema looks entirely independent of the shape of the data rows, so can you just expose a `pub fn encode_transaction` method in this module?

(The `transaction_id` parameter for the encode pipeline makes sense though.)